failed to compile on mac os

I tried to install lightGBM on mac and failed to compile it. 

`
Scanning dependencies of target lightgbm
Scanning dependencies of target _lightgbm
[  4%] Building CXX object CMakeFiles/lightgbm.dir/src/main.cpp.o
[  4%] Building CXX object CMakeFiles/_lightgbm.dir/src/c_api.cpp.o
[  4%] Building CXX object CMakeFiles/lightgbm.dir/src/application/application.cpp.o
[  6%] Building CXX object CMakeFiles/_lightgbm.dir/src/lightgbm_R.cpp.o
[  7%] Building CXX object CMakeFiles/_lightgbm.dir/src/application/application.cpp.o
[  9%] Building CXX object CMakeFiles/lightgbm.dir/src/boosting/boosting.cpp.o
[ 11%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/boosting.cpp.o
[ 12%] Building CXX object CMakeFiles/lightgbm.dir/src/boosting/gbdt.cpp.o
[ 14%] Building CXX object CMakeFiles/lightgbm.dir/src/boosting/gbdt_model_text.cpp.o
[ 15%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt.cpp.o
[ 17%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt_model_text.cpp.o
[ 19%] Building CXX object CMakeFiles/lightgbm.dir/src/boosting/gbdt_prediction.cpp.o
[ 20%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/gbdt_prediction.cpp.o
[ 22%] Building CXX object CMakeFiles/lightgbm.dir/src/boosting/prediction_early_stop.cpp.o
[ 23%] Building CXX object CMakeFiles/_lightgbm.dir/src/boosting/prediction_early_stop.cpp.o
[ 25%] Building CXX object CMakeFiles/lightgbm.dir/src/io/bin.cpp.o
[ 26%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/bin.cpp.o
[ 28%] Building CXX object CMakeFiles/lightgbm.dir/src/io/config.cpp.o
[ 30%] Building CXX object CMakeFiles/lightgbm.dir/src/io/config_auto.cpp.o
[ 31%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/config.cpp.o
[ 33%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/config_auto.cpp.o
[ 34%] Building CXX object CMakeFiles/lightgbm.dir/src/io/dataset.cpp.o
[ 36%] Building CXX object CMakeFiles/lightgbm.dir/src/io/dataset_loader.cpp.o
[ 38%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/dataset.cpp.o
[ 39%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/dataset_loader.cpp.o
[ 41%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/file_io.cpp.o
[ 42%] Building CXX object CMakeFiles/lightgbm.dir/src/io/file_io.cpp.o
[ 44%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/json11.cpp.o
[ 46%] Building CXX object CMakeFiles/lightgbm.dir/src/io/json11.cpp.o
[ 47%] Building CXX object CMakeFiles/lightgbm.dir/src/io/metadata.cpp.o
[ 49%] Building CXX object CMakeFiles/lightgbm.dir/src/io/parser.cpp.o
[ 50%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/metadata.cpp.o
[ 52%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/parser.cpp.o
[ 53%] Building CXX object CMakeFiles/_lightgbm.dir/src/io/tree.cpp.o
[ 55%] Building CXX object CMakeFiles/lightgbm.dir/src/io/tree.cpp.o
[ 57%] Building CXX object CMakeFiles/lightgbm.dir/src/metric/dcg_calculator.cpp.o
[ 58%] Building CXX object CMakeFiles/lightgbm.dir/src/metric/metric.cpp.o
[ 60%] Building CXX object CMakeFiles/_lightgbm.dir/src/metric/dcg_calculator.cpp.o
[ 61%] Building CXX object CMakeFiles/lightgbm.dir/src/network/linker_topo.cpp.o
[ 63%] Building CXX object CMakeFiles/_lightgbm.dir/src/metric/metric.cpp.o
[ 65%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/linker_topo.cpp.o
[ 66%] Building CXX object CMakeFiles/lightgbm.dir/src/network/linkers_mpi.cpp.o
[ 68%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/linkers_mpi.cpp.o
[ 69%] Building CXX object CMakeFiles/lightgbm.dir/src/network/linkers_socket.cpp.o
[ 71%] Building CXX object CMakeFiles/lightgbm.dir/src/network/network.cpp.o
[ 73%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/linkers_socket.cpp.o
[ 74%] Building CXX object CMakeFiles/_lightgbm.dir/src/network/network.cpp.o
[ 76%] Building CXX object CMakeFiles/lightgbm.dir/src/objective/objective_function.cpp.o
[ 77%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/data_parallel_tree_learner.cpp.o
[ 79%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/feature_parallel_tree_learner.cpp.o
[ 80%] Building CXX object CMakeFiles/_lightgbm.dir/src/objective/objective_function.cpp.o
[ 82%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/data_parallel_tree_learner.cpp.o
[ 84%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/feature_parallel_tree_learner.cpp.o
[ 85%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/gpu_tree_learner.cpp.o
[ 87%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/gpu_tree_learner.cpp.o
[ 88%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/serial_tree_learner.cpp.o
[ 90%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/tree_learner.cpp.o
[ 92%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/serial_tree_learner.cpp.o
[ 93%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/tree_learner.cpp.o
[ 95%] Building CXX object CMakeFiles/lightgbm.dir/src/treelearner/voting_parallel_tree_learner.cpp.o
[ 96%] Building CXX object CMakeFiles/_lightgbm.dir/src/treelearner/voting_parallel_tree_learner.cpp.o
[ 98%] Linking CXX executable ../lightgbm
Undefined symbols for architecture x86_64:
  "_GOMP_atomic_end", referenced from:
      LightGBM::RegressionL2loss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::Init(LightGBM::Metadata const&, int) [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropy::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropyLambda::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::LeafSplits::Init(float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      LightGBM::LeafSplits::Init(int, LightGBM::DataPartition const*, float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      ...
  "_GOMP_atomic_start", referenced from:
      LightGBM::RegressionL2loss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::Init(LightGBM::Metadata const&, int) [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropy::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropyLambda::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::LeafSplits::Init(float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      LightGBM::LeafSplits::Init(int, LightGBM::DataPartition const*, float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      ...
  "_GOMP_loop_end_nowait", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_loop_nonmonotonic_guided_next", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_loop_nonmonotonic_guided_start", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_parallel", referenced from:
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) in application.cpp.o
      std::_Function_handler<void (int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&), LightGBM::Predictor::Predict(char const*, char const*, bool)::{lambda(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)#2}>::_M_invoke(std::_Any_data const&, int&&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) in application.cpp.o
      LightGBM::Application::Predict()     in application.cpp.o
      LightGBM::GBDT::InitPredict(int, bool) in boosting.cpp.o
      LightGBM::RF::AddValidDataset(LightGBM::Dataset const*, std::vector<LightGBM::Metric const*, std::allocator<LightGBM::Metric const*> > const&) in boosting.cpp.o
      LightGBM::RF::RollbackOneIter()     in boosting.cpp.o
      LightGBM::GOSS::Bagging(int) in boosting.cpp.o
      ...
  "_omp_get_num_threads", referenced from:
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) [clone ._omp_fn.0] in application.cpp.o
      LightGBM::Predictor::Predict(char const*, char const*, bool)::{lambda(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)#2}::operator()(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) const [clone ._omp_fn.0] in application.cpp.o
      LightGBM::Application::Predict() [clone ._omp_fn.0] in application.cpp.o
      LightGBM::GBDT::InitPredict(int, bool) [clone ._omp_fn.0] in boosting.cpp.o
      LightGBM::Tree::Shrinkage(double) [clone ._omp_fn.0] in boosting.cpp.o
      LightGBM::GOSS::Bagging(int) [clone ._omp_fn.1] in boosting.cpp.o
      LightGBM::RF::Boosting() [clone ._omp_fn.0] in boosting.cpp.o
      ...
  "_omp_get_thread_num", referenced from:
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) [clone ._omp_fn.0] in application.cpp.o
      std::_Function_handler<void (std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*), LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double)::{lambda(std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*)#2}>::_M_invoke(std::_Any_data const&, std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*&&) in application.cpp.o
      LightGBM::Predictor::Predict(char const*, char const*, bool)::{lambda(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)#2}::operator()(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) const [clone ._omp_fn.0] in application.cpp.o
      LightGBM::Application::Predict() [clone ._omp_fn.0] in application.cpp.o
      std::_Function_handler<void (std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*), LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double)::{lambda(std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*)#4}>::_M_invoke(std::_Any_data const&, std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*&&) in application.cpp.o
      std::_Function_handler<void (std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*), LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double)::{lambda(std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*)#3}>::_M_invoke(std::_Any_data const&, std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*&&) in application.cpp.o
      std::_Function_handler<void (std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*), LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double)::{lambda(std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*)#1}>::_M_invoke(std::_Any_data const&, std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*&&) in application.cpp.o
      ...
  "_omp_set_nested", referenced from:
      LightGBM::Application::Application(int, char**) in application.cpp.o
  "_omp_set_num_threads", referenced from:
      LightGBM::Application::Application(int, char**) in application.cpp.o
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [../lightgbm] Error 1
make[1]: *** [CMakeFiles/lightgbm.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[100%] Linking CXX shared library ../lib_lightgbm.so
Undefined symbols for architecture x86_64:
  "_GOMP_atomic_end", referenced from:
      LightGBM::RegressionL2loss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::Init(LightGBM::Metadata const&, int) [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropy::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropyLambda::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::LeafSplits::Init(float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      LightGBM::LeafSplits::Init(int, LightGBM::DataPartition const*, float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      ...
  "_GOMP_atomic_start", referenced from:
      LightGBM::RegressionL2loss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::BinaryLogloss::Init(LightGBM::Metadata const&, int) [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropy::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::CrossEntropyLambda::BoostFromScore(int) const [clone ._omp_fn.0] in objective_function.cpp.o
      LightGBM::LeafSplits::Init(float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      LightGBM::LeafSplits::Init(int, LightGBM::DataPartition const*, float const*, float const*) [clone ._omp_fn.0] in serial_tree_learner.cpp.o
      ...
  "_GOMP_loop_end_nowait", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_loop_nonmonotonic_guided_next", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_loop_nonmonotonic_guided_start", referenced from:
      LightGBM::Dataset::FinishLoad() [clone ._omp_fn.0] in dataset.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::CostructFromSampleData(double**, int**, int, int const*, unsigned long, int) [clone ._omp_fn.1] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.0] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.2] in dataset_loader.cpp.o
      LightGBM::DatasetLoader::ConstructBinMappersFromTextData(int, int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&, LightGBM::Parser const*, LightGBM::Dataset*) [clone ._omp_fn.1] in dataset_loader.cpp.o
      ...
  "_GOMP_parallel", referenced from:
      _LGBM_DatasetPushRows in c_api.cpp.o
      _LGBM_DatasetPushRowsByCSR in c_api.cpp.o
      std::_Function_handler<void (int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&), LightGBM::Predictor::Predict(char const*, char const*, bool)::{lambda(int, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&)#2}>::_M_invoke(std::_Any_data const&, int&&, std::vector<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::allocator<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > > const&) in c_api.cpp.o
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) in c_api.cpp.o
      LightGBM::Booster::Predict(int, int, int, std::function<std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > (int)>, LightGBM::Config const&, double*, long long*) in c_api.cpp.o
      _LGBM_DatasetCreateFromCSR in c_api.cpp.o
      _LGBM_DatasetCreateFromCSRFunc in c_api.cpp.o
      ...
  "_omp_get_num_threads", referenced from:
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) [clone ._omp_fn.0] in c_api.cpp.o
      _LGBM_BoosterPredictForCSC._omp_fn.0 in c_api.cpp.o
      LightGBM::Booster::Predict(int, int, int, std::function<std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > (int)>, LightGBM::Config const&, double*, long long*) [clone ._omp_fn.0] in c_api.cpp.o
      _LGBM_DatasetPushRows._omp_fn.0 in c_api.cpp.o
      _LGBM_DatasetCreateFromMats._omp_fn.0 in c_api.cpp.o
      _LGBM_DatasetPushRowsByCSR._omp_fn.0 in c_api.cpp.o
      _LGBM_DatasetCreateFromCSR._omp_fn.0 in c_api.cpp.o
      ...
  "_omp_get_thread_num", referenced from:
      LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double) [clone ._omp_fn.0] in c_api.cpp.o
      _LGBM_BoosterPredictForCSC._omp_fn.0 in c_api.cpp.o
      std::_Function_handler<void (std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*), LightGBM::Predictor::Predictor(LightGBM::Boosting*, int, bool, bool, bool, bool, int, double)::{lambda(std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*)#2}>::_M_invoke(std::_Any_data const&, std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > const&, double*&&) in c_api.cpp.o
      LightGBM::Booster::Predict(int, int, int, std::function<std::vector<std::pair<int, double>, std::allocator<std::pair<int, double> > > (int)>, LightGBM::Config const&, double*, long long*) [clone ._omp_fn.0] in c_api.cpp.o
      _LGBM_DatasetPushRows._omp_fn.0 in c_api.cpp.o
      _LGBM_DatasetCreateFromMats._omp_fn.0 in c_api.cpp.o
      _LGBM_DatasetPushRowsByCSR._omp_fn.0 in c_api.cpp.o
      ...
  "_omp_set_nested", referenced from:
      LightGBM::Application::Application(int, char**) in application.cpp.o
  "_omp_set_num_threads", referenced from:
      _LGBM_DatasetCreateFromFile in c_api.cpp.o
      _LGBM_DatasetCreateFromSampledColumn in c_api.cpp.o
      _LGBM_BoosterPredictForCSR in c_api.cpp.o
      _LGBM_BoosterPredictForMat in c_api.cpp.o
      _LGBM_BoosterPredictForMats in c_api.cpp.o
      _LGBM_BoosterPredictForCSRSingleRow in c_api.cpp.o
      _LGBM_BoosterPredictForMatSingleRow in c_api.cpp.o
      ...
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[2]: *** [../lib_lightgbm.so] Error 1
make[1]: *** [CMakeFiles/_lightgbm.dir/all] Error 2
make: *** [all] Error 2

`

Does anyone know how to solve this error?

--- a/R-package/src/install.libs.R
+++ b/R-package/src/install.libs.R
@@ -45,8 +45,16 @@ if (!use_precompile) {
   }
 
   # Could NOT find OpenMP_C on Mojave workaround
-  using_gcc <- grepl('gcc', Sys.getenv('CC', ''))
-  using_gpp <- grepl('g\\+\\+', Sys.getenv('CXX', ''))
+  # Using this kind-of complicated pattern to avoid matching to
+  # things like "pgcc"
+  using_gcc <- grepl(
+    pattern = '^gcc$|[/\\]+gcc|^gcc\\-[0-9]+$|[/\\]+gcc\\-[0-9]+$'
+    , x = Sys.getenv('CC', '')
+  )
+  using_gpp <- grepl(
+    pattern = '^g\\+\\+$|[/\\]+g\\+\\+|^g\\+\\+\\-[0-9]+$|[/\\]+g\\+\\+\\-[0-9]+$'
+    , x = Sys.getenv('CXX', '')
+  )
   on_mac <- Sys.info()['sysname'] == 'Darwin'
   if (on_mac && !(using_gcc & using_gpp)) {
     cmake_cmd <- paste(cmake_cmd, ' -DOpenMP_C_FLAGS="-Xpreprocessor -fopenmp -I$(brew --prefix libomp)/include" ')

--- a/R-package/src/install.libs.R
+++ b/R-package/src/install.libs.R
@@ -45,7 +45,10 @@ if (!use_precompile) {
   }
 
   # Could NOT find OpenMP_C on Mojave workaround
-  if (Sys.info()['sysname'] == 'Darwin' && !(grepl('^gcc', Sys.getenv('CC', '')) & grepl('^g\\+\\+', Sys.getenv('CXX', '')))) {
+  using_gcc <- grepl('gcc', Sys.getenv('CC', ''))
+  using_gpp <- grepl('g\\+\\+', Sys.getenv('CXX', ''))
+  on_mac <- Sys.info()['sysname'] == 'Darwin'
+  if (on_mac && !(using_gcc & using_gpp)) {
     cmake_cmd <- paste(cmake_cmd, ' -DOpenMP_C_FLAGS="-Xpreprocessor -fopenmp -I$(brew --prefix libomp)/include" ')
     cmake_cmd <- paste(cmake_cmd, ' -DOpenMP_C_LIB_NAMES="omp" ')
     cmake_cmd <- paste(cmake_cmd, ' -DOpenMP_CXX_FLAGS="-Xpreprocessor -fopenmp -I$(brew --prefix libomp)/include" ')

